subsys/mpsl/pin_debug/mpsl_pin_debug_nrf54.c

```diff
@@ -0,0 +1,229 @@
+#include <nrfx_dppi.h>
+#include <hal/nrf_radio.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+
+const nrfx_gpiote_t gpiote = NRFX_GPIOTE_INSTANCE(20);
+const nrfx_dppi_t dppi_radio_domain = NRFX_DPPI_INSTANCE(10);
+const nrfx_dppi_t dppi_gpio_domain = NRFX_DPPI_INSTANCE(20);
+LOG_MODULE_REGISTER(mpsl_radio_pin_debug, CONFIG_MPSL_LOG_LEVEL);
+
+static int m_ppi_config(void)
+{
+	uint8_t gppi_chan_radio_ready;
+	uint8_t gppi_chan_radio_disabled;
+	uint8_t gppi_chan_radio_address;
+	uint8_t gppi_chan_radio_end;
+
+	uint8_t dppi_chan_gpio_ready;
+	uint8_t dppi_chan_gpio_disabled;
+	uint8_t dppi_chan_gpio_address;
+	uint8_t dppi_chan_gpio_end;
+
+	if (nrfx_gppi_channel_alloc(&gppi_chan_radio_ready) !=
+	    NRFX_SUCCESS) {
+		LOG_ERR("Failed allocating gppi_chan_radio_ready");
+		return -ENOMEM;
+	}
+
+	if (nrfx_gppi_channel_alloc(&gppi_chan_radio_disabled) !=
+	    NRFX_SUCCESS) {
+		LOG_ERR("Failed allocating gppi_chan_radio_disabled");
+		return -ENOMEM;
+	}
+
+	if (nrfx_gppi_channel_alloc(&gppi_chan_radio_address) !=
+	    NRFX_SUCCESS) {
+		LOG_ERR("Failed allocating gppi_chan_radio_address");
+		return -ENOMEM;
+	}
+
+	if (nrfx_gppi_channel_alloc(&gppi_chan_radio_end) !=
+	    NRFX_SUCCESS) {
+		LOG_ERR("Failed allocating gppi_chan_radio_end");
+		return -ENOMEM;
+	}
+
+	if (nrfx_dppi_channel_alloc(&dppi_gpio_domain, &dppi_chan_gpio_ready) !=
+	    NRFX_SUCCESS) {
+		LOG_ERR("Failed allocating dppi_chan_gpio_ready");
+		return -ENOMEM;
+	}
+
+	if (nrfx_dppi_channel_alloc(&dppi_gpio_domain, &dppi_chan_gpio_disabled) !=
+	    NRFX_SUCCESS) {
+		LOG_ERR("Failed allocating dppi_chan_gpio_disabled");
+		return -ENOMEM;
+	}
+
+	if (nrfx_dppi_channel_alloc(&dppi_gpio_domain, &dppi_chan_gpio_address) !=
+	    NRFX_SUCCESS) {
+		LOG_ERR("Failed allocating dppi_chan_gpio_address");
+		return -ENOMEM;
+	}
+
+	if (nrfx_dppi_channel_alloc(&dppi_gpio_domain, &dppi_chan_gpio_end) !=
+	    NRFX_SUCCESS) {
+		LOG_ERR("Failed allocating dppi_chan_gpio_end");
+		return -ENOMEM;
+	}
+
+	if (nrfx_gppi_edge_connection_setup(gppi_chan_radio_ready,
+					    &dppi_radio_domain,
+					    MPSL_DPPI_RADIO_PUBLISH_READY_CHANNEL_IDX,
+					    &dppi_gpio_domain,
+					    dppi_chan_gpio_ready) != NRFX_SUCCESS) {
+		LOG_ERR("Failed edge setup chan ready");
+		return -ENOMEM;
+	}
+
+	if (nrfx_gppi_edge_connection_setup(gppi_chan_radio_disabled,
+					    &dppi_radio_domain,
+					    MPSL_DPPI_RADIO_PUBLISH_DISABLED_CH_IDX,
+					    &dppi_gpio_domain,
+					    dppi_chan_gpio_disabled) != NRFX_SUCCESS) {
+		LOG_ERR("Failed edge setup chan disabled");
+		return -ENOMEM;
+	}
+
+	if (nrfx_gppi_edge_connection_setup(gppi_chan_radio_address,
+					    &dppi_radio_domain,
+					    MPSL_DPPI_RADIO_PUBLISH_ADDRESS_CHANNEL_IDX,
+					    &dppi_gpio_domain,
+					    dppi_chan_gpio_address) != NRFX_SUCCESS) {
+		LOG_ERR("Failed edge setup chan address");
+		return -ENOMEM;
+	}
+
+	/* Setup a PPI bridge between the radio domain and the domain of the GPIO pin. */
```

Comment:
Consider to have comment like this before each of the calls to `nrfx_gppi_edge_connection_setup`